### DISCUSSION TITLE
Structured vs. unstructured grids

### DISCUSSION THREAD

**Discussion by aiskhak** [idaholab/moose#19346] [Q&A Meshing] [answered]
Hello,

I am doing simulations using MOOSE-based SAM for 3D coarse-grid thermal-hydraulics and I am facing the following issue.
The code works fine with unstructured grids, though, I need to have a structured grid for my framework. As soon as I develop structured grids, the code considerably slows down (figures of grids attached). One of the possible issues is that structured grids inevitably have sharp transitions and I am not sure that MOOSE (FEM) can handle this appropriately. Maybe it is possible to fix this by using different preconditioners and stuff like that?

```
[Preconditioning]
  [./SMP_PJFNK]
    type = SMP
    full = true
    solve_type = 'PJFNK'
    petsc_options_iname = '-pc_type -ksp_gmres_restart'
    petsc_options_value = 'lu 100'
  [../]
[]
```

![2021-11-10_10-16-41](https://user-images.githubusercontent.com/62344361/141140146-f98739c5-76d1-44a8-8737-15004d6e8583.png)

![2021-11-10_10-18-48](https://user-images.githubusercontent.com/62344361/141140394-cae31b86-471e-4689-851c-a29fcd714fbf.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

What does this geometry look like in 3D? Is it square like this still? 
If so you should really use a quad / hex mesh.
This second mesh looks incredibly skewed and with high aspect ratios. I dont think you will get a good answer from MOOSE or SAM there. 

The cannon fluid flow solver in MOOSE are the finite volume fluid flow capabilities in the Navier Stokes module, and the FE in that module are supported as well. We dont know what was implemented in SAM. We are getting good results with structured & unstructured grids with finite volume. 

A better numerical scheme, including a better preconditioner, would indeed help you get a solution. We are currently implementing segregated solver capabilities for the finite volume solver. 

Guillaume

**Reply by aiskhak** [reply]
The mesh in 3D attached (the square region with round inlet-outlet tubes):
![image](https://user-images.githubusercontent.com/62344361/141161438-6a22c0e6-6dc4-438e-8606-ee9ad463ff9a.png)

Yes, the unstructured grid looks bad, but it gives much-much better results in terms of convergence.

SAM has only FEM implementation. Though, I am not sure how would I use the finite volume approach...
Are there any recommendation of what to try with the FEM?
